### ISSUE TITLE
python3 repo salt-api pkg has incorrect shebang line

### ISSUE THREAD

**Issue by brandon-sling** [saltstack/salt#55940]
### Description of Issue
http://repo.saltstack.com/py3/ubuntu/16.04/amd64/2019.2/
salt-api pkg lists it's shebang as /usr/bin/python instead of /usr/bin/python3

### Setup
install salt-master, salt-minion, salt-ssh from http://repo.saltstack.com/py3/ubuntu/16.04/amd64/2019.2/

### Steps to Reproduce Issue
install salt-api from http://repo.saltstack.com/py3/ubuntu/16.04/amd64/2019.2/
The salt-api install will fail to start with an error about not finding the module salt.scripts

### Fix
modifying /usr/bin/salt-api and adding a 3 to the end of the shebang line allows salt-api to start

**Comment by brandon-sling** [comment]
I suspect this also applies to the package in http://repo.saltstack.com/py3/ubuntu/16.04/amd64/latest too since they look to have been uploaded at the same time

**Comment by dmurphy18** [comment]
@brandon-sling Having installed Salt 2019.2.3 Py 3 into an Ubuntu 16.04 system, the contents of salt-api where as follows:

root@david-VirtualBox:/home/david# cat /usr/bin/salt-api 
#!/usr/bin/python3

# Import salt libs
from salt.scripts import salt_api


if __name__ == '__main__':
    salt_api()
root@david-VirtualBox:/home/david# 

root@david-VirtualBox:/home/david# lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.2 LTS
Release:	16.04
Codename:	xenial
root@david-VirtualBox:/home/david#

Could you detail the steps used to install salt-api, please check if the issue is upgrading from a Python 2 installation to Python 3 and the Python 2 files are not being overwritten, possibly due to a 'N' in a Y/N query.

**Comment by brandon-sling** [comment]
@dmurphy18 It was an upgrade from a python2 installation on my master.
I originally tried to use the bootstrap script with `-P -M -x python3`. This caused issues because not everything from salt was upgraded to python3.
I then `apt-get remove`'d all installed salt-* packages followed by `apt autoremove`
Since the box wasn't a salt master at this point I modified the sources.list.d file to point at the py3 repo instead of the py2 and ran apt-get update. Checked `apt-cache policy salt-*` had the correct version, and installed salt-master, salt-minion, salt-ssh, and salt-api.
This is when I noticed /usr/bin/salt-api didn't have python3 in the shebang. I removed it and installed again but still said python instead of python3. So I just manually modified the file and made this ticket

**Comment by dmurphy18** [comment]
@brandon-sling Thanks for the additional information,, it looks like an upgrade issue from Python2 to Python 3.  Will investigate it as such.

**Comment by brandon-sling** [comment]
@dmurphy18 I think you're right about this. I was looking and I still have a 'salt' package installed according to `python -m pip list`. (My python command still points to 2.7.12) None of the other boxes where I did an upgrade on the salt-minion have a salt package still installed in the python2.7 directory. This was a master not a minion but still.

**Comment by dmurphy18** [comment]
@brandon-sling What was the output of the 'python -m pip list', curious to know.

**Comment by brandon-sling** [comment]
```
root@d-gp2-salt-2:~# python -m pip list
Package         Version               
--------------- ----------------------
beautifulsoup4  4.4.1                 
chardet         2.3.0                 
cryptography    1.2.3                 
enum34          1.1.2                 
futures         3.0.5                 
html5lib        0.999                 
idna            2.0                   
iotop           0.6                   
ipaddress       1.0.16                
Jinja2          2.8                   
lxml            3.5.0                 
M2Crypto        0.22.6rc4             
MarkupSafe      0.23                  
msgpack-python  0.4.6                 
mysqlclient     1.3.7                 
ndg-httpsclient 0.4.0                 
pip             10.0.1                
psutil          3.4.2                 
psycopg2        2.6.1                 
pyasn1          0.1.9                 
pycrypto        2.6.1                 
pycurl          7.43.0                
PyMySQL         0.7.2                 
pyOpenSSL       0.15.1                
python-apt      1.1.0b1+ubuntu0.16.4.5
PyYAML          3.11                  
pyzmq           15.2.0                
requests        2.9.1                 
salt            2018.3.4              
setuptools      20.7.0                
six             1.10.0                
smmap           0.9.0                 
tornado         4.2.1                 
urllib3         1.13.1                
wheel           0.29.0                
Zenoss-Plugins  2.0.4
```

**Comment by brandon-sling** [comment]
Also I removed everything again and made sure to use `apt-get purge` and `apt autoremove` and cleared apt-cache and reinstalled all the packages and the `/usr/bin/salt-api` shows python3 in the shebang after installation now. So there is probably still something in the upgrade of python2 to python3 that might catch someone but I guess this can be considered resolved.

PS: pip won't let me remove a lot of those packages because they were installed by dist-utils. So I'm guessing the first time I did the upgrade the package data was removed from the system, but the files got left.
